src/ja/common/job.py

```diff
@@ -39,58 +39,103 @@ def __init__(self,
                  special_resources: List[str]):
-        pass
+        self._priority = priority
+        self._is_preemptible = is_preemtible
+        self._special_resources = special_resources
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, JobSchedulingConstraints):
+            return self.priority == other.priority \
+                and self.is_preemptible == other.is_preemptible \
+                and self.special_resources == other.special_resources
+        else:
+            return False
 
     @property
     def priority(self) -> JobPriority:
         """!
         @return The priority of the job.
         """
+        return self._priority
 
     @property
     def is_preemptible(self) -> bool:
         """!
         @return Whether the job can be preempted.
         """
+        return self._is_preemptible
 
     @property
-    def get_special_resources(self) -> List[str]:
+    def special_resources(self) -> List[str]:
         """!
         @return A list of special resources used by the job.
         """
+        return self._special_resources
 
     def to_dict(self) -> Dict[str, object]:
-        pass
+        _dict: Dict[str, object] = dict()
+        _dict["priority"] = self.priority
+        _dict["is_preemptible"] = self.is_preemptible
+        _dict["special_resources"] = self.special_resources
+        return _dict
 
     @classmethod
     def from_dict(cls, property_dict: Dict[str, object]) -> "JobSchedulingConstraints":
-        pass
+        _priority = JobPriority(cls._get_from_dict(property_dict=property_dict, key="priority"))
+        _is_preemtible = cls._get_bool_from_dict(property_dict=property_dict, key="is_preemptible")
+        _special_resources = cls._get_str_list_from_dict(property_dict=property_dict, key="special_resources")
+
+        cls._assert_all_properties_used(property_dict)
+
+        return JobSchedulingConstraints(
+            priority=_priority, is_preemtible=_is_preemtible, special_resources=_special_resources)
 
 
 class Job(Serializable):
     """!
     Represents a job with all of its attributes.
     @param owner_id The unix user id of the user who owns this job.
     @param email The email of the user who created the job, or None.
-    @param ctx The docker context of the job.
-    @param constraints The docker constraints of the job.
+    @param scheduling_constraints The docker constraints of the job.
+    @param docker_context The docker context of the job.
     @param label The label set by the user for the job.
     """
     def __init__(self,
                  owner_id: int,
                  email: str,
-                 constraints: JobSchedulingConstraints,
-                 ctx: IDockerContext,
+                 scheduling_constraints: JobSchedulingConstraints,
+                 docker_context: IDockerContext,
                  docker_constraints: DockerConstraints,
                  label: str = None):
-        pass
+        self._uid: str = None
```

Comment:
How will we generate the uid, i fount [this ](https://docs.python.org/3/library/uuid.html) online, but maybe we can just start with 1 and increment for each new job.

Reply:
> "If the job has not been queued yet, it has no UID." aa I see.